install_config/router/customized_haproxy_router.adoc

```diff
@@ -13,48 +13,476 @@ toc::[]
+|`*Disable*` | no traffic is allowed on the insecure port.
+|`*Redirect*` | clients are redirected to the secure port.
+|===
+None ("") is the same as `*Disable*`.
+
+[[using-annotations]]
+=== Annotations
+
+Each route can have annotations attached. The Annotations are just a name
+and a value.
+
+[source,yaml]
+----
+apiVersion: v1
+kind: Route
+metadata:
+  annotations:
+    haproxy.router.openshift.io/timeout: 5500ms
+[...]
+----
+
+The name can be pretty much anything that doesn't conflict with existing
+Annotations.  The value is any string. The string can have multiple tokens
+separated by a space e.g., "aa bb cc".  The template uses {{index}} to
+extract the value of an annotation. For example:
+
+----
+{{$balanceAlgo := index $cfg.Annotations "haproxy.router.openshift.io/balance"}}
+----
+
+See .xref:../../architecture/core_concepts/routes.adoc#route-specific-annotations[Route-specific Annotations]
+for more information.
+
+
+[[using-env-var]]
+=== Environment Variables
+
+The template can use whatever environment variables exist in the router pod.
+The environment variables can be set in the deployment configuration. Any
+of the existing environment variables can be used and new ones can be added.
+
+They are referenced by the env function:
+
+----
+{{env "ROUTER_MAX_CONNECTIONS" "20000"}}
+----
+
+Where the first string is the variable, and the second string is the default
+when the variable is missing or *nil*. When `ROUTER_MAX_CONNECTIONS` is not
+set or is *nil*, 20000 is used. Environment variables are a map where the key
+is the environment variable name and the content is the value of the string.
+
+See .xref:../../architecture/core_concepts/routes.adoc#env-variables[Route-specific Environment variables]
+for more information.
+
+
+[[example]]
+
+=== Pulling It All Together, An Example
+
+Here is a simplistic template based on the HAProxy template file.
+
+First off lets start with a comment:
+
+----
+{{/*
+  Here is a small example of how to work with templates
+  taken from the haproxy template file
+*/}}
+----
+
+Lets start to create an output file. The define construct gives the
+file name and everything up to the closing "end" contributes to the file.
+
+----
+{{ define "/var/lib/haproxy/conf/haproxy.config" }}
+global
+{{ end }}
+----
+
+The above will copy *global* to the */var/lib/haproxy/conf/haproxy.config* file
+and then close the file.
+
+Lets set up logging based on environment variables.
+
+----
+{{ with (env "ROUTER_SYSLOG_ADDRESS" "") }}
+  log {{.}} {{env "ROUTER_LOG_FACILITY" "local1"}} {{env "ROUTER_LOG_LEVEL" "warning"}}
+{{ end }}
+----
+
+The environment variables are a map with the environment variable name as
+the key and the contents as the value. The *env* function extracts the value
+for the environment variable.  if the environment variable is not defined or
+*nil* the second argument is returned.
+
+The the result of evaluating what is between the "()" goes in "." (dot).
```

Comment:
Redundant "the".

This is a little confusing.  How about, "The `with` construct sets the value of `.` (dot) within the `with` block to whatever value is provided as an argument to `with`."?